python/lsst/db/mysqlUtils.py

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2014-2015 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+This module contains utilities / helpers that perform common tasks related to
+MySQL. This includes:
+ * reading mysql credential files
+ * managing database and tables (create, drop, check if exists, listing, checking
+   if a table is a view)
+ * checking if user exists
+ * loading SQL scripts
```

Comment:
It looks like another utility of this module is to wrap sqlalchemy exceptions in our own exceptions type. I'm not sure that this is the right thing to do in all cases. Majority of the use cases will be direct use of sqlalchemy classes (like running SELECT or INSERT queries) and those will generate sqlalchemy exceptions. methods in this module will generate exceptions of different type. It may be confusing or complicated to mix sqlalchemy methods with methods from this module and try to handle exceptions from both.